webrender_traits/src/api.rs

```diff
@@ -301,21 +303,23 @@ impl RenderApi {
                             background_color: Option<ColorF>,
                             epoch: Epoch,
                             viewport_size: LayoutSize,
-                            (pipeline_id, display_list): (PipelineId, BuiltDisplayList),
+                            display_list_builder: DisplayListBuilder,
                             preserve_frame_state: bool) {
-        let (dl_data, dl_desc) = display_list.into_data();
+        let (pipeline_id, content_size, display_list) = display_list_builder.finalize();
```

Comment:
This breaks Gecko. We finalize in a different process from the one that calls set_display_list. Is there a particular reason for this?